### PR TITLE
book: Document encoding of proofs

### PR DESCRIPTION
**Pull request by str4d**
Closes zcash/halo2#144.

### CODE REVIEW

File: book/src/design/implementation/proofs.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+- PLONK evaluations:
+  - $(2 + \textsf{Col}_P) \times P$ scalars (repeated $N$ times).
+  - $5L$ scalars (repeated $N$ times).
+
+- Multiopening argument:
+  - 1 point.
+  - 1 scalar per set of points in the multiopening argument.
+
+- Polynomial commitment scheme:
+  - $1 + 2k$ points.
```
Comment (therealyingtong) [inline book/src/design/implementation/proofs.md:87]:
```suggestion
  - $1 + 2k$ points, where $k$ is the number of rounds in the inner product argument.
```

Comment (str4d) [reply book/src/design/implementation/proofs.md:87]: `k` is also the size of the circuit (2<sup>k</sup> rows), so I'll call this out as a constant further up.

File: book/src/design/implementation/proofs.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+- $F$ - the number of fixed columns.
+- $I$ - the number of instance columns.
+- $L$ - the number of lookup arguments.
+- $P$ - the number of permutation arguments.
+- $\textsf{Col}_P$ - the number of columns involved in permutation argument $P$.
+- $D$ - the maximum degree for the quotient polynomial.
+- $Q_A$ - the number of advice column queries.
+- $Q_F$ - the number of fixed column queries.
+- $Q_I$ - the number of instance column queries.
+- $N$ - the number of instances of the circuit that are being proven simultaneously.
```
Comment (ebfull) [inline book/src/design/implementation/proofs.md:61]: I'd prefer to use `M` here because `n` is defined often in the code as `2^k`.

**Review by ebfull — APPROVED** [review]
ACK modulo nit. A few parts of this (namely the permutation argument and vanishing argument parts) will be changing, but those changes can happen in the PR that accompanies the code changes.

File: book/src/design/implementation/proofs.md (hunk 2 above)
Comment (str4d) [reply book/src/design/implementation/proofs.md:61]: Fine by me!

**Pull request merged**